### PR TITLE
Re-deprecate `--start-time-[cpu-]us` flags 

### PR DESCRIPTION
**Pull request by roypat**
## Changes

Deprecates the `--start-time-cpu-us` and `--start-time-us` CLI arguments and adds a `DEPRECATIONS.md` file.

## Reason

printing the deprecation message caused an observable change in behavior, so the best we can do is "deprecation via changelog".

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this
  PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by zulinx86 — APPROVED** [review]

File: DEPRECATED.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+a future major Firecracker release, in accordance with our
+[release policy](docs/RELEASE_POLICY.md).
+
+- \[[#2763](https://github.com/firecracker-microvm/firecracker/pull/2763)\] The
+  `vsock_id` body field in `PUT` requests on `/vsock`
+- \[[#2980](https://github.com/firecracker-microvm/firecracker/pull/2980)\] The
+  `mem_file_path` body field in `PUT` requests on `/snapshot/load`
+- \[[#2973](https://github.com/firecracker-microvm/firecracker/pull/2973)\]
+  MicroVM Metadata Service v1 (MMDSv1)
+- \[[#4126](https://github.com/firecracker-microvm/firecracker/pull/4126)\] The
```
Comment (pb8o) [inline DEPRECATED.md:13]: And static CPU templates in general also no? Those could be used if starting from a JSON file.

Comment (roypat) [reply DEPRECATED.md:13]: Mh, no sure. @zulinx86 ?

Comment (zulinx86) [reply DEPRECATED.md:13]: Yes, we'll drop the support of static CPU templates in general (not limited to API but also FC JSON config).

Comment (roypat) [reply DEPRECATED.md]: Updated, thanks!

**Review by zulinx86 — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**